### PR TITLE
fixed adding an event to favourites

### LINKED ISSUE #5702: [Event preview card ] Authorized User isn't able to add an event to favourite
**Environment:** Windows 10 Pro, Chrome 111.0.5563.65
**Reproducible:** always
**Build found:** 23.04.2023

**Preconditions**
1. Opened https://www.testgreencity.ga/#/events
2. Have more than one event

**Steps to reproduce**
1. Click on the flag button on the Events card
2. Refresh the page

**Actual result**
The settings of the selected event have not been saved

**Expected result**
Selected event (added to favourite) is shown in the top of events list

**User story and test case links**
E.g.: "View event preview card #3374
[Test case](https://jira.softserve.academy/browse/GC-5164)"

[Video reproduce bug](https://monosnap.com/file/KWwSDhbVgeu0eWjLoo5z0m3GpuUZQp)

### PR DESCRIPTION
**Pull request by igorv404**
# GreenCityClient PR
[#5702](https://github.com/ita-social-projects/GreenCity/issues/5702)

## Summary Of Changes :fire:
fixed adding an event to favourites

## Added
* updated function of adding event to favourites
* added function of removing event from favourites
* tests for new functions

## How to test :clipboard:
Repeat actions in following request: [#5702](https://github.com/ita-social-projects/GreenCity/issues/5702)
# PR Checklist Forms

- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [ ] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts [hunk 1]
```diff
@@ -157,6 +157,13 @@ export class ProfileDashboardComponent implements OnInit, OnDestroy {
       });
   }
 
+  removeUnFavouriteEvent(id: number): void {
+    const indexOfUnFavouriteEvent = this.favouriteEvents.findIndex((event) => event.id === id);
+    if (indexOfUnFavouriteEvent !== -1) {
+      this.favouriteEvents.splice(indexOfUnFavouriteEvent, 1);
+    }
+  }
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts:162]: I would use filter method here
 removeUnFavouriteEvent(id: number): void {
    this.favouriteEvents = this.favouriteEvents.filter((event) => event.id !== id);
  }

File: src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts [hunk 2]
```diff
@@ -157,6 +157,13 @@ export class ProfileDashboardComponent implements OnInit, OnDestroy {
...
+
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts:163]: also you need to fix unit tests

File: src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts (hunk 1 above)
Comment (igorv404) [reply src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts:162]: Done it

File: src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts (hunk 2 above)
Comment (igorv404) [reply src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts:163]: Fixed

**Review by yurikhrunyk — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Review by sobkiv — APPROVED** [review]

**Review by OksanaFilipchuk — APPROVED** [review]

**Review by ShyraiDan — APPROVED** [review]

**Pull request merged**